src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -151,11 +166,41 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
 
   try {
     checkTezosConnection()
-    mainLoop(0)
+    val accountsRefreshEpochs = Await.result(unprocessedEpochsForRefreshingAccounts(), atMost = 5.seconds)
+    mainLoop(0, SortedSet(accountsRefreshEpochs: _*))
   } finally {
     shutdown()
   }
 
+  /* Possibly updates all accounts if the current block level is past any of the epochs
+   * @param epochs the relevant epochs that calls for a refresh
+   */
+  private def processAccountRefreshes(epochs: SortedSet[Int]): Future[SortedSet[Int]] =
+    if (epochs.nonEmpty) {
+      for {
+        storedHead <- apiOperations.fetchMaxLevel
+        updated <- if (epochs.exists(_ <= storedHead)) {
+          val (past, toCome) = epochs.partition(_ <= storedHead)
+          logger.info(
+            "A block level was reached requiring to update the accounts data. Relevant block levels are {}",
```

Comment:
There should also be a log message printed when the refresh is complete.

Reply:
I will add